### DISCUSSION TITLE
VLESS-HTTP2-TLS 组合形式也可以用。

### DISCUSSION THREAD

**Discussion by chika0801** [XTLS/Xray-core#2873] [General]
服务端回落参数是可选，个人推荐加吧。由Xray监听443，不能启用Vision流控，但是有http2带来的复用buff。参考gRPC用了这么久，个人认为不会很快遇到被封端口，需要自己实验（不负责）

`"network": "h2",` 这里写成 `http` 和写 `h2` 是一样的效果。以前这个组合没人提，那时http/h2不快，现在REALITY出来h2已经修过了，好起来了。

不要忘记用这个隐藏参数。
```json
                "httpSettings": {
                    "host": [] // 若不写此字段或值留空时，将使用默认值 "www.example.com"，若设置此字段，需要两端值一致才能连接成功，"host": [""] 不是值留空
                }
```

服务端入站部分
```
        {
            "listen": "0.0.0.0",
            "port": 443,
            "protocol": "vless",
            "settings": {
                "clients": [
                    {
                        "id": "chika"
                    }
                ],
                "decryption": "none"
            },
            "streamSettings": {
                "network": "h2",
                "security": "tls",
                "tlsSettings": {
                    "rejectUnknownSni": true,
                    "minVersion": "1.2",
                    "certificates": [
                        {
                            "ocspStapling": 3600,
                            "certificateFile": "/etc/ssl/private/fullchain.cer",
                            "keyFile": "/etc/ssl/private/private.key"
                        }
                    ]
                }
            },
            "sniffing": {
                "enabled": true,
                "destOverride": [
                    "http",
                    "tls",
                    "quic"
                ]
            }
        }

```

客户端出站部分
```
    "outbounds": [
        {
            "protocol": "vless",
            "settings": {
                "vnext": [
                    {
                        "address": "xxxxxxxxxxxxxxxxxx",
                        "port": 443,
                        "users": [
                            {
                                "id": "chika",
                                "encryption": "none"
                            }
                        ]
                    }
                ]
            },
            "streamSettings": {
                "network": "h2",
                "security": "tls",
                "tlsSettings": {
                    "serverName": "xxxxxxxxxxxxxxxxx",
                    "fingerprint": "chrome"
                },
                "sockopt": {
                    "tcpMptcp": true,
                    "tcpNoDelay": true
                }
            }
        }
    ]
```

**Comment by chise0713** [comment]
> ~~使用jsonc作为反引号 “ ` ```jsonc ` ” 的标识，在注释时不会出现红色提示~~
```jsonc
{
    "test": "test"
} // comment
```

**Reply by chika0801** [reply]
这个我知道的，那故意没用jsonc

**Reply by chise0713** [reply]
> 这个我知道的，那故意没用jsonc

~~我的~~

**Reply by chika0801** [reply]
有段历史是当时gRPC才出来时（Xray版本更新时）

文档上写的 https://xtls.github.io/Xray-docs-next/config/transports/grpc.html 建议是用Nginx监听443。再代给Xray。当时也有人试过直接Xray监听443 gRPC（带不带TLS忘记了） 协议用VMESS还是SS。我印象中没见最后说端口被封没有的报告。

比如现在这里提到的用法，Xray监听443，VLESS+TLS+HTTP2。假设使用VLESS的回落到nginx反代一个网站。我不确定如何先假设GFW有主动探测，这种组合被封端口的机率大不大。

同理那为什么不也试试Xray监听443，VLESS+TLS+gRPC都有回落在。

可能这组合当时因为HTTP2不快也没人再拿出来说。REALITY来的时候RPRX把HTTP2不快也修了，就再想起它来了。

**Comment by chika0801** [comment]
对的，这组合tls指纹是go（xray）的。服务端go指纹也是由不良林有个视频带来的热度，至于gfw拿它当不当什么我们也不知道。

有复用带来缓解的tls in tls，感谢你的使用体验。

自定义padding好像有xray的switch有关，现在鸽了。

**Comment by cwyin7788** [comment]
> 服务端回落参数是可选，个人推荐加吧。由Xray监听443，不能启用Vision流控，但是有http2带来的复用buff。参考gRPC用了这么久，个人认为不会很快遇到被封端口，需要自己实验（不负责）
> 
> 
> 
> `"network": "h2",` 这里写成 `http` 和写 `h2` 是一样的效果。以前这个组合没人提，那时http/h2不快，现在REALITY出来h2已经修过了，好起来了。
> 
> 
> 
> 不要忘记用这个隐藏参数。
> 
> ```json
> 
>                 "httpSettings": {
> 
>                     "host": [] // 若不写此字段或值留空时，将使用默认值 "www.example.com"，若设置此字段，需要两端值一致才能连接成功，"host": [""] 不是值留空
> 
>                 }
> 
> ```
> 
> 
> 
> 服务端入站部分
> 
> ```
> 
>         {
> 
>             "listen": "0.0.0.0",
> 
>             "port": 443,
> 
>             "protocol": "vless",
> 
>             "settings": {
> 
>                 "clients": [
> 
>                     {
> 
>                         "id": "chika"
> 
>                     }
> 
>                 ],
> 
>                 "decryption": "none"
> 
>             },
> 
>             "streamSettings": {
> 
>                 "network": "h2",
> 
>                 "security": "tls",
> 
>                 "tlsSettings": {
> 
>                     "rejectUnknownSni": true,
> 
>                     "minVersion": "1.2",
> 
>                     "certificates": [
> 
>                         {
> 
>                             "ocspStapling": 3600,
> 
>                             "certificateFile": "/etc/ssl/private/fullchain.cer",
> 
>                             "keyFile": "/etc/ssl/private/private.key"
> 
>                         }
> 
>                     ]
> 
>                 }
> 
>             },
> 
>             "sniffing": {
> 
>                 "enabled": true,
> 
>                 "destOverride": [
> 
>                     "http",
> 
>                     "tls",
> 
>                     "quic"
> 
>                 ]
> 
>             }
> 
>         }
> 
> 
> 
> ```
> 
> 
> 
> 客户端出站部分
> 
> ```
> 
>     "outbounds": [
> 
>         {
> 
>             "protocol": "vless",
> 
>             "settings": {
> 
>                 "vnext": [
> 
>                     {
> 
>                         "address": "xxxxxxxxxxxxxxxxxx",
> 
>                         "port": 443,
> 
>                         "users": [
> 
>                             {
> 
>                                 "id": "chika",
> 
>                                 "encryption": "none"
> 
>                             }
> 
>                         ]
> 
>                     }
> 
>                 ]
> 
>             },
> 
>             "streamSettings": {
> 
>                 "network": "h2",
> 
>                 "security": "tls",
> 
>                 "tlsSettings": {
> 
>                     "serverName": "xxxxxxxxxxxxxxxxx",
> 
>                     "fingerprint": "chrome"
> 
>                 },
> 
>                 "sockopt": {
> 
>                     "tcpMptcp": true,
> 
>                     "tcpNoDelay": true
> 
>                 }
> 
>             }
> 
>         }
> 
>     ]
> 
> ```

我成功配置好了，但这个不是用vision，应该没有回落的吧，那它是否不能配置伪装网站呢？谢谢

**Reply by lxhao61** [reply]
可以配置伪装网站，用 Caddy 做代理及提供 TLS。

**Comment by chika0801** [comment]
> 应该没有回落的吧 

没有。

> 那它是否不能配置伪装网站呢？

~~要么你试试Xray服务端配置中你配下回落，再用nginx接对应端口。~~
测试了，结果是失败

要么回到传统的nginx监听443传递到xray的形式。

要不要回落，防不防主动探测，你认为有主动探测且让你VPS端口IP被封，信则加，不信随便。。。
